File: src/client_gen/go/debug.go
package main

import . "project_name/model"
import "bufio"

type Debug struct {
	Writer *bufio.Writer
}

func (debug Debug) Send(command DebugCommand) {
	ClientMessageDebugMessage {
		Command: command,
	}.Write(debug.Writer)
	err := debug.Writer.Flush()
	if err != nil {
		panic(err)
	}
}

File: src/client_gen/javascript/debug.js
const model = require('./model/index');

class Debug {
    constructor(streamWrapper) {
        this.streamWrapper = streamWrapper;
    }

    async send(command) {
        await (new model.ClientMessage.DebugMessage(command)).writeTo(this.streamWrapper);
        // TODO: only flush stream once here?
    }
}

module.exports.Debug = Debug;